components/vr-video.js

```diff
@@ -0,0 +1,63 @@
+/* globals define */
+(function(define){'use strict';define(function(require,exports,module){
+
+  document.registerElement(
+    'vr-video',
+    {
+      prototype: Object.create(
+        VRObject.prototype, {
+          init: {
+            value: function() {
+              var material = this.getMaterial();
+              var geometry = this.getGeometry();
+              this.object3D = new THREE.Mesh(geometry, material);
+              this.load();
+            }
+          },
+
+          update: {
+            value: function() {
+              var material = this.getMaterial();
+              var geometry = this.getGeometry();
+              this.object3D = new THREE.Mesh(geometry, material);
+              this.object3D.material = material;
+            }
+          },
+
+          getGeometry: {
+            value: function() {
+              var width = parseFloat(this.getAttribute('width')) || 50;
+              var height = parseFloat(this.getAttribute('height')) || 50;
+              return new THREE.PlaneGeometry(width, height, 1, 1);
+            }
+          },
+
+          getMaterial: {
+            value: function() {
+              var video = document.createElement('video');
+              video.crossOrigin = 'anonymous';
+              video.src = this.getAttribute('src');;
+              video.autoplay = this.hasAttribute('autoplay');
+              video.loop = this.hasAttribute('loop');
```

Comment:
for binary attributes, it might be better to check hasAttribute *and* that getAttribute is not explicitly `false`